src/jaxsim/parsers/rod/meshes.py

```diff
@@ -0,0 +1,194 @@
+
+    Args:
+        mesh: The mesh from which to extract points.
+        axis: The axis along which to extract points.
+        direction: The direction along the axis from which to extract points. Valid values are "higher" and "lower".
+        n: The number of points to extract.
+
+    Returns:
+        The extracted points (N x 3 array).
+    """
+
+    valid_dirs = ["higher", "lower"]
+    if direction not in valid_dirs:
+        raise ValueError(f"Invalid direction. Valid directions are {valid_dirs}")
+
+    arr = mesh.vertices
+
+    index = 0 if axis == "x" else 1 if axis == "y" else 2
+    # Sort the array in ascending order
+    sorted_arr = arr[arr[:, index].argsort()]
+
+    if direction == "lower":
+        # Select first N points
+        points = sorted_arr[:n]
+    elif direction == "higher":
+        # Select last N points
+        points = sorted_arr[-n:]
+    else:
+        raise ValueError(
+            f"Invalid direction {direction} for SelectPointsOverAxis method"
+        )
+
+    return points
+
+
+def extract_points_app(
+    mesh: trimesh.Trimesh,
+    axis: str,
+    upper: float | None = None,
+    lower: float | None = None,
+) -> np.ndarray:
+    """
+    Extracts points from a mesh along a specified axis within a specified range. The points are selected based on their position along the axis.
+
+    Args:
+        mesh: The mesh from which to extract points.
+        axis: The axis along which to extract points.
+        upper: The upper bound of the range.
+        lower: The lower bound of the range.
+
+    Returns:
+        The extracted points (N x 3 array).
+
+    Raises:
+        AssertionError: If the lower bound is greater than the upper bound.
+    """
+
+    # Check bounds
+    upper = upper if upper is not None else np.inf
+    lower = lower if lower is not None else -np.inf
+    assert lower < upper, "Invalid bounds for axis-aligned plane"
+
+    # Logic
+    valid_axes = {"x": 0, "y": 1, "z": 2}
+
+    points = mesh.vertices[
+        (mesh.vertices[:, valid_axes[axis]] >= lower)
+        & (mesh.vertices[:, valid_axes[axis]] <= upper)
+    ]
+
+    return points
+
+
+def extract_points_object_mapping(
+    mesh: trimesh.Trimesh,
+    objs: Sequence[trimesh.Trimesh | dict],
+    method: str = "subtract",
+) -> np.ndarray:
+    """
+    Extracts points from a mesh by mapping objects onto it.
+
+    Args:
+        mesh: The mesh from which to extract points.
+        objs: The objects to map onto the mesh.
+        method: The method to use for object mapping. Valid values are "subtract" and "intersect".
+
+    Returns:
+        The extracted points (N x 3 array).
+
+    Raises:
+        ValueError: If an invalid method is provided.
+    """
+
+    valid_methods = ["subtract", "intersect"]
+    if method not in valid_methods:
+        raise ValueError(f"Invalid method {method} for object mapping")
```

Comment:
Try to use a dictionary as above, this case will be automatically covered by a KeyError